### PR TITLE
fix(TwabRewards): check create and cancel inputs

### PR DESCRIPTION
**Pull request by PierrickGT**
Issue: https://github.com/code-423n4/2021-12-pooltogether-findings/issues/29

### CODE REVIEW

File: contracts/TwabRewards.sol [hunk 1]
```diff
@@ -157,8 +160,9 @@ contract TwabRewards is ITwabRewards {
         override
         returns (bool)
     {
-        Promotion memory _promotion = _getPromotion(_promotionId);
+        _requireNumberOfEpochs(_numberOfEpochs);
```
Comment (asselstine) [inline contracts/TwabRewards.sol:163]: This isn't quite correct here.  Need to check that the passed _numberOfEpochs is less than (256 - currentNumberOfEpochs)

File: contracts/TwabRewards.sol [hunk 2]
```diff
@@ -257,6 +261,14 @@ contract TwabRewards is ITwabRewards {
         require(succeeded && controllerAddress != address(0), "TwabRewards/invalid-ticket");
     }
 
+    /**
+        @notice Allow a promotion to be created or extended only by a positive number of epochs.
+        @param _numberOfEpochs Number of epochs to check
+    */
+    function _requireNumberOfEpochs(uint8 _numberOfEpochs) internal view {
+        require(_numberOfEpochs > 0, "TwabRewards/epochs-not-zero");
```
Comment (asselstine) [inline contracts/TwabRewards.sol:269]: might as well pull this up into the createPromo, once the above refactoring is made

**Review by asselstine — CHANGES_REQUESTED** [review]

File: contracts/TwabRewards.sol (hunk 1 above)
Comment (PierrickGT) [reply contracts/TwabRewards.sol:163]: It should be enforced by the `uint8` type no?
It should overflow on the following line: https://github.com/pooltogether/v4-periphery/blob/22094acf51b7645c78c668bab0c2059f045a61f6/contracts/TwabRewards.sol#L168

Comment (asselstine) [reply contracts/TwabRewards.sol:147]: Relying on overflow works, but isn't the best for the user.  They won't know what broke!

File: contracts/TwabRewards.sol (hunk 2 above)
Comment (PierrickGT) [reply contracts/TwabRewards.sol:250]: It should still revert if we don't actually extend the promotion when passing 0.

File: contracts/TwabRewards.sol (hunk 1 above)
Comment (PierrickGT) [reply contracts/TwabRewards.sol:147]: Done: https://github.com/pooltogether/v4-periphery/pull/34/commits/1f6a4064cce58d4bf6ee7dd8bfa47698d2ce12e1

**Pull request merged**